src/util_libc.rs

```diff
@@ -5,12 +5,28 @@
 // <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
-extern crate std;
-
 use crate::util::LazyUsize;
 use crate::Error;
+use core::num::NonZeroU32;
 use core::ptr::NonNull;
-use std::io;
+
+cfg_if! {
+    if #[cfg(any(target_os = "netbsd", target_os = "openbsd", target_os = "android"))] {
+        use libc::__errno as errno_location;
+    } else if #[cfg(any(target_os = "linux", target_os = "emscripten", target_os = "redox"))] {
+        use libc::__errno_location as errno_location;
+    } else if #[cfg(any(target_os = "solaris", target_os = "illumos"))] {
+        use libc::___errno as errno_location;
+    } else if #[cfg(any(target_os = "macos", target_os = "freebsd", target_os = "dragonfly"))] {
+        use libc::__error as errno_location;
+    } else if #[cfg(target_os = "haiku")] {
+        use libc::_errnop as errno_location;
+    }
+}
+
+pub fn last_os_error() -> Error {
+    Error::from(NonZeroU32::new(unsafe { *errno_location() } as u32).unwrap())
```

Comment:
Potential panic point. Maybe write something like:
 ```rust
let errno = unsafe { *errno_location() } as u32;
NonZeroU32::new(errno).map_or(Error::from, Error::ZERO_ERRNO)
```
?

Reply:
I did something slightly different, so that we would also catch if the `errno` is negative. The compiler now can [figure everything out, and remove the `unwrap()` code](https://rust.godbolt.org/z/TeEtmN).